feat: add new screen for incorrect balance

yongjiajie opened the pull request:
[Notion link](https://www.notion.so/sally-wallet/Error-handling-when-Sally-calls-GovWallet-v2-customer-account-get-for-MINDEF-SAFRA-campaign-f442fe104e0b49bd931998d034b5755e?v=532cc4d27b0641d69c5f34e3f6f0ab78) <!-- Remove this link if no relevant Notion link -->

This PR adds... <!-- A brief description of what your PR does -->

- `useGovWalletBalance` hook now returns balance of first account in `govWalletBalanceInCents`.
- `formatCentsAsDollarsAndCents`, a new utility function that converts `number` amount in cents to `string` amount in dollars and cents for display.
- `IncorrectBalanceCard` which is shown when customer has quota, has an activated account in GovWallet, but has the incorrect balance.
- Translations for `IncorrectBalanceCard` in `en` and `zh`
- Added new state for `useGovWalletBalance` hook, `FETCHING_BALANCE`, to ensure checkout button is still loading whilst balance is being fetched to prevent users from checking out too early.

<!-- This checklist is just a guideline.-->
<!-- If any of the following does not apply to your PR, please leave them unchecked and explain in the PR description -->

- [x] I've kept this PR as small as possible (~600 lines) by splitting it into PRs with manageable chunks of code
- [x] I've requested reviews from 1-2 reviewers
- [x] I've added [accessibility IDs](https://www.notion.so/dc5f4ce910f7431c84344ac79344e9f5?v=d487366816834dd4ab8dc12e0b5928f3) to my components
- [x] I've added/updated unit/integration tests
- [ ] I've added/updated [e2e tests](https://www.notion.so/e2e-Documentation-a096d3e0bf75485e85ad692af8371ef3) for at least the happy flow
- [x] I've added [Chinese translations for i18n setup](https://www.notion.so/Translations-Dev-Guide-8c1da838982a4f59a386ec96ac6780c8)
- [ ] I've added documentation in README/[Notion](https://www.notion.so/82b92fb1007640328dab9582c0a3694e?v=3b6ce48202cf40ad8450553799b13146)

auronsiow commented on src/utils/currencyFormatter.ts:
```diff
@@ -0,0 +1,22 @@
+  amountInCents: number,
+  separator = "."
+): string => {
+  const amountString = amountInCents.toString();
```
Can be simplified with a small change.
`return (amountInCents / 100).toLocaleString("en-US", { minimumFractionDigits: 2, });`.
The test you've written should still pass. 😁

yongjiajie replied:
This is done in [1aff2ff](https://github.com/rationally-app/mobile-application/pull/609/commits/1aff2ff755e0ba3455c50ce1fa13c1a01f527891)!

auronsiow reviewed (dismissed):
Looks good!

yongjiajie commented:
@auronsiow 

Added the following commits after finding some issues with the text: 
- [4793e0c](https://github.com/rationally-app/mobile-application/pull/609/commits/4793e0c521a94153b52f07a80cad0eefc62fefa8)
- [e0f5f0a](https://github.com/rationally-app/mobile-application/pull/609/commits/e0f5f0a00959dab9b631855e6b3aa8c25352390a)
- [e55bc8e](https://github.com/rationally-app/mobile-application/pull/609/commits/e55bc8e320925ab6ac9a4e34edaa461e6fc2e9ec)
- [3a46adc](https://github.com/rationally-app/mobile-application/pull/609/commits/3a46adcf36b2d85102db74820668c8c1b1ec600c)

Found another issue where the user is able to click "Check Out" while their balance is being checked, since the states does not prevent the user from clicking on that button. 

Added another state to allow us to control when the "Check Out" button can be pressed, in the following commits:
- [1d3be03](https://github.com/rationally-app/mobile-application/pull/609/commits/1d3be03a94f366eed087721fed9132bc76cfa23d)
- [53e82a2](https://github.com/rationally-app/mobile-application/pull/609/commits/53e82a2a115babfa8196ccf2bad2306f06884fd6)
- [27826b3](https://github.com/rationally-app/mobile-application/pull/609/commits/27826b332617442450a8534b1fd2126ee2168244)

auronsiow reviewed (approved):
Looks good to me 👍
